Palindrome.UnitTests.cs

```diff
@@ -0,0 +1,117 @@
+        public void Check_Longest_Palindrome_In_Dictionary()
+        {
+            Palindrome check = new Palindrome();
+            // The longest palindromic word in the Oxford English Dictionary is "tattarrattat", 
+            // coined by James Joyce in Ulysses (1922) for a knock on the door.
+            char[] arrToCheck = check.PalindromeHelper("tattarrattat");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(true, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Odd_Palindrome_One()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("radar");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(true, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Odd_Palindrome_Two()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("catytac");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(true, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Even_Palindrome_One()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("Otto");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(true, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Even_Palindrome_Two()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("Hannah");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(true, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Non_Palindrome_Outer_Chars_Do_Not_Match()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("aooooooob");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(false, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Palindrome_Inner_Chars_Do_Not_Match()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("oooabooo");
+
+            bool boolResultOfPalindromeCheck = check.IsArrayPalindrome(arrToCheck);
+
+            Assert.AreEqual(false, boolResultOfPalindromeCheck);
+        }
+        [TestMethod]
+        public void Check_Sentence_Palindrome()
+        {
+            Palindrome check = new Palindrome();
+            char[] arrToCheck = check.PalindromeHelper("Murder for a jar of red rum");
```

Comment:
If you want to support this scenario, I would update the `IsArrayPalindrome` method to accept a `string` with code like this:

```csharp
public bool IsArrayPalindrome(string input)
{
    if (string.isNullOrWhitespace(input))
    {
        throw new ArgumentNullException(nameof(input));
    }
    
    char[] values = input.Trim().Replace(" ", string.Empty).ToCharArray();
    ...
}
```
You don't really need your helper method.